### PR TITLE
Basic Refactoring

### PR DESCRIPTION
**Pull request by antrix190**
Added Strings to constants and reused it elsewhere.

### CODE REVIEW

File: src/main/java/me/amarpandey/config/SchedulerConfig.java [hunk 1]
```diff
@@ -17,6 +19,6 @@
 
 	@Scheduled(fixedDelay = 3000)
 	public void sendAdhocMessages() {
-		template.convertAndSend("/topic/stats", count);
+    template.convertAndSend(Constants.TOPIC_STATS, count);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/config/SchedulerConfig.java]: make `TOPIC_STATS` a static import

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 2]
```diff
@@ -17,20 +17,20 @@
...
 
 import me.amarpandey.model.UserResponse;
+import me.amarpandey.utils.Constants;
 
 @Controller
 public class ChatController {
 
 	@MessageMapping("/connect")
-	@SendTo("/topic/connect")
+  @SendTo(Constants.TOPIC_CONNECT)
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: same here. make it static import

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 3]
```diff
@@ -17,20 +17,20 @@
...
 	public UserResponse connect(@RequestParam String username, SimpMessageHeaderAccessor headerAccessor) {
 		headerAccessor.getSessionAttributes().put("username", username);
-		UserResponse userResponse = new UserResponse(username, NEW_USER_JOINED, JOIN, PUBLIC);
-		return userResponse;
+    return new UserResponse(username, NEW_USER_JOINED, JOIN, PUBLIC);
 	}
 
 	@MessageMapping("/message")
-	@SendTo("/topic/message")
+  @SendTo(Constants.TOPIC_MESSAGE)
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: here also

File: src/main/java/me/amarpandey/event/WebSocketEventListener.java [hunk 4]
```diff
@@ -39,12 +40,12 @@ public void handleWebsocketDisconnectListener(SessionDisconnectEvent event) {
 		count = count > 0 ? count - 1 : 0;
 
 		SimpMessageHeaderAccessor headerAccessor = StompHeaderAccessor.wrap(event.getMessage());
-		String username = (String) headerAccessor.getSessionAttributes().get("username");
+    String username = (String) headerAccessor.getSessionAttributes().get(Constants.USERNAME);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/event/WebSocketEventListener.java]: here also

File: src/main/java/me/amarpandey/event/WebSocketEventListener.java [hunk 5]
```diff
@@ -39,12 +40,12 @@ public void handleWebsocketDisconnectListener(SessionDisconnectEvent event) {
...
 
 		if (username != null) {
 
 			UserResponse userResponse = new UserResponse(username, USER_LEFT, LEAVE, PUBLIC);
-			simpMessagingTemplate.convertAndSend("/topic/message", userResponse);
+      simpMessagingTemplate.convertAndSend(Constants.TOPIC_MESSAGE, userResponse);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/event/WebSocketEventListener.java]: here also

File: src/main/java/me/amarpandey/config/SchedulerConfig.java (hunk 1 above)
Comment (antrix190) [reply src/main/java/me/amarpandey/config/SchedulerConfig.java]: Its defined in Contants.java
Why do you want to specifically import in each file?

Comment (amarlearning) [reply src/main/java/me/amarpandey/config/SchedulerConfig.java]: there is no need to import `Constants` when we need only `TOPIC_STATS`.

Inshort, use 
```java
import static me.amarpandey.utils.Constants.TOPIC_STATS;
```
Instead
```java
import me.amarpandey.utils.Constants;
```

**Comment by antrix190** [comment]
@amarlearning Done.

**Comment by amarlearning** [comment]
@antrix190 some improper indentation I can see, please correct that.

**Comment by antrix190** [comment]
Can you point me to the part you're referring to?

File: src/main/java/me/amarpandey/config/SchedulerConfig.java [hunk 6]
```diff
@@ -17,6 +18,6 @@
 
 	@Scheduled(fixedDelay = 3000)
 	public void sendAdhocMessages() {
-		template.convertAndSend("/topic/stats", count);
+    template.convertAndSend(TOPIC_STATS, count);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/config/SchedulerConfig.java]: indentation problem here

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 7]
```diff
@@ -22,15 +25,14 @@
 public class ChatController {
 
 	@MessageMapping("/connect")
-	@SendTo("/topic/connect")
+  @SendTo(TOPIC_CONNECT)
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: indentation problem

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 8]
```diff
@@ -22,15 +25,14 @@
...
 	public UserResponse connect(@RequestParam String username, SimpMessageHeaderAccessor headerAccessor) {
-		headerAccessor.getSessionAttributes().put("username", username);
-		UserResponse userResponse = new UserResponse(username, NEW_USER_JOINED, JOIN, PUBLIC);
-		return userResponse;
+    headerAccessor.getSessionAttributes().put(USERNAME, username);
+    return new UserResponse(username, NEW_USER_JOINED, JOIN, PUBLIC);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: indentation problem

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 9]
```diff
@@ -22,15 +25,14 @@
...
 	}
 
 	@MessageMapping("/message")
-	@SendTo("/topic/message")
+  @SendTo(TOPIC_MESSAGE)
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: indentation problem

File: src/main/java/me/amarpandey/controller/ChatController.java [hunk 10]
```diff
@@ -44,9 +46,8 @@ public UserResponse getMessage(@Payload UserResponse userResponse) {
 	@SendTo("/topic/{groupid}/connect")
 	public UserResponse connect(@DestinationVariable String groupid, @RequestParam String username,
 			SimpMessageHeaderAccessor headerAccessor) {
-		headerAccessor.getSessionAttributes().put("username", username);
-		UserResponse userResponse = new UserResponse(username, NEW_USER_JOINED, JOIN, PRIVATE);
-		return userResponse;
+    headerAccessor.getSessionAttributes().put(USERNAME, username);
+    return new UserResponse(username, NEW_USER_JOINED, JOIN, PRIVATE);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/controller/ChatController.java]: indentation problem

File: src/main/java/me/amarpandey/event/WebSocketEventListener.java [hunk 11]
```diff
@@ -30,7 +32,7 @@
 	@EventListener(SessionConnectEvent.class)
 	public void handleWebsocketConnectListener(SessionConnectEvent event) {
 		count = count + 1;
-		logger.info("Received a new web socket connection : " + now());
+    logger.info("Received a new web socket connection : " + now());
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/event/WebSocketEventListener.java]: indentation problem

File: src/main/java/me/amarpandey/event/WebSocketEventListener.java [hunk 12]
```diff
@@ -39,12 +41,12 @@ public void handleWebsocketDisconnectListener(SessionDisconnectEvent event) {
 		count = count > 0 ? count - 1 : 0;
 
 		SimpMessageHeaderAccessor headerAccessor = StompHeaderAccessor.wrap(event.getMessage());
-		String username = (String) headerAccessor.getSessionAttributes().get("username");
+    String username = (String) headerAccessor.getSessionAttributes().get(USERNAME);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/event/WebSocketEventListener.java]: indentation problem

File: src/main/java/me/amarpandey/event/WebSocketEventListener.java [hunk 13]
```diff
@@ -39,12 +41,12 @@ public void handleWebsocketDisconnectListener(SessionDisconnectEvent event) {
...
 
 		if (username != null) {
 
 			UserResponse userResponse = new UserResponse(username, USER_LEFT, LEAVE, PUBLIC);
-			simpMessagingTemplate.convertAndSend("/topic/message", userResponse);
+      simpMessagingTemplate.convertAndSend(TOPIC_MESSAGE, userResponse);
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/event/WebSocketEventListener.java]: indentation problem

File: src/main/java/me/amarpandey/utils/Constants.java [hunk 14]
```diff
@@ -5,4 +5,11 @@
 	public static final String USER_LEFT = "Left!";
 	public static final String NEW_USER_JOINED = "Joined!";
 
+  public static final String TOPIC_MESSAGE = "/topic/message";
+  public static final String TOPIC_CONNECT = "/topic/connect";
+  public static final String TOPIC_STATS = "/topic/stats";
+
+  public static final String USERNAME = "username";
+
+
```
Comment (amarlearning) [inline src/main/java/me/amarpandey/utils/Constants.java]: indentation problem from line 8 - 12 and remove empty line

**Pull request closed without merging**